data/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { createStore } from 'redux';
+import { flowRight } from 'lodash';
+
+/**
+ * Module constants
+ */
+const reducers = [];
+const enhancers = [];
+if ( window.__REDUX_DEVTOOLS_EXTENSION__ ) {
+	enhancers.push( window.__REDUX_DEVTOOLS_EXTENSION__() );
+}
+const store = createStore( dynamicReducer, {}, flowRight( enhancers ) );
+
+/**
+ * Reducer function combining the dynamic "reducers" array
+ *
+ * @param {Object} state  Global state
+ * @param {Object} action Action
+ *
+ * @return {Object}       Updated global state
+ */
+function dynamicReducer( state = {}, action ) {
+	let hasChanges = false;
+	const newState = {};
+	reducers.forEach( ( { key, reducer } ) => {
+		const newSubState = reducer( state[ key ] || {}, action );
+		hasChanges = hasChanges || newSubState !== state[ key ];
+		newState[ key ] = newSubState;
+	} );
+
+	return hasChanges ? newState : state;
+}
+
+/**
+ * Register a new sub reducer to the global sate
+ *
+ * @param {String} key     Reducer key
+ * @param {Object} reducer Reducer function
+ */
+export function registerReducer( key, reducer ) {
+	reducers.push( { key, reducer } );
+	store.dispatch( { type: 'NEW_REDUCER', key, reducer } );
```

Comment:
Looking at this makes me wonder if `dynamicReducer` could be implemented to maintain reducers as its own state, rather than assigning reducers to the top-level variable. Maybe needless complexity.

Reply:
I actually thought about it and opted for a global for simplicity.

Reply:
Have you explored [replaceReducer](https://redux.js.org/docs/api/Store.html#replaceReducer) method offered by Redux? This could be combined with combineReducers. I don't know what limitation it has though.

Reply:
Oh! thanks for sharing that. I think I read about it before but forgot it completely. I don't expect it to change the performance or the API but maybe it can save us some lines of code.